packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -301,22 +300,6 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
     }
   });
 
-  // Focus on mouse down/touch up to match native textfield behavior.
-  // usePress handles canceling text selection.
-  let {pressProps} = usePress({
```

Comment:
Removing usePress here fixes the behavior with usePreventScroll, which needs to control how focus is managed on iOS when tapping an editable element.